mlibrary/lib.umich.edu
# Website 116 social media sharing fix

# Overview
The method for sharing pages on social media has changed since the site was created. [Facebook now relies on Open Graph markup](https://developers.facebook.com/docs/sharing/webmasters#testing) as `<meta>` tags in the site's `<head>` to gather relevant information.

From the linked documentation, it states "server must also use the gzip and deflate encodings". I'm not entirely sure what this means, but it sounds like a lot of potential back-end work.

For now, I'd like to test the new `<meta>` tags and open graph properties to see if Facebook is properly crawling the pages again. I'm hoping to not have to rely on a plugin to help me get sharing working again, but if needed, there are [options we can explore](https://www.gatsbyjs.com/plugins/gatsby-remark-opengraph/?=open%20graph%20sharing).

This is phase 1 of  [WEBSITE-114](https://mlit.atlassian.net/jira/software/projects/WEBSITE/boards/27?selectedIssue=WEBSITE-116). This PR is just testing Facebook sharing functionality. I can not test the sharing on the staging site since Facebook can not crawl staging, only prod. If the new `<meta>` tags are not working as intended and Facebook continues to show the lib homepage preview, we may want to revert these changes and explore other options. If it works, I'll continue implementing code for our Twitter sharing as well.

## Anything else?
While developing this code, I was getting build errors on pages without location data as part of the GraphQL query. The first page that returned a build error in this instance was:

/visit-and-study/cafes-and-wellbeing/reflection-rooms

It should be fixed by using the "useLocation" hook and location check here:
`return (siteData.site.siteMetadata.siteUrl + location?.pathname`

On the most recent staging build it looks to be working as intended. These errors happen in the build phase, but it may be worthwhile to take a closer look at this page along with a news / event page with sharing options enabled.

## Testing
- Make sure the PR is consistent in these browsers:
- Run accessibility tests:
- How to use the feature
- [x] test the /visit-and-study/cafes-and-wellbeing/reflection-rooms page
- [x] **On Previews** take a look at an event / news page with sharing enabled. For example: /visit-and-study/events-and-exhibits/today-and-upcoming/souq-stories-gaza-lives
- [ ] **On Prod (once merged)** take at an event / news page with sharing enabled. For example: /visit-and-study/events-and-exhibits/today-and-upcoming/souq-stories-gaza-lives
